raiden-dapp/src/components/AmountInput.vue

```diff
@@ -125,22 +104,80 @@ export default class AmountInput extends Vue {
     }
 
     const value = clipboardData.getData('text');
-    if (!AmountInput.numericRegex.test(value)) {
+
+    if (!this.numericRegEx.test(value)) {
       event.preventDefault();
     } else {
       const input = event.target as HTMLInputElement;
       input.setSelectionRange(0, input.value.length);
     }
   }
 
-  onInput(value: string) {
-    /* istanbul ignore else */
-    if (this.$refs.input) {
-      const input = this.$refs.input as any;
-      this.valid = input.valid;
+  updateIfValid(value: string) {
+    this.errorMessages.pop();
+
+    switch (false) {
+      case this.noValidInput(value):
+        this.errorMessages.push(
+          this.$t('amount-input.error.invalid') as string
+        );
+        break;
+      case this.noDecimalOverflow(value):
+        this.errorMessages.push(
+          this.$t('amount-input.error.too-many-decimals', {
+            decimals: this.token!.decimals
+          }) as string
+        );
+        break;
+      case this.parsedAmount(value):
+        this.errorMessages.push(this.$t('amount-input.error.zero') as string);
+        break;
+      case this.hasEnoughBalance(value, this.max):
+        this.errorMessages.push(
+          this.$t('amount-input.error.not-enough-funds', {
+            funds: BalanceUtils.toUnits(this.max, this.token!.decimals ?? 18),
+            symbol: this.token!.symbol
+          }) as string
+        );
+        break;
+      default:
+        if (value !== this.amount && !Number.isNaN(Number(value))) {
+          this.amount = value;
+        }
     }
+  }
 
-    this.$emit('input', value);
+  noValidInput(value: string): boolean {
+    return !Number.isNaN(Number(value));
+  }
```

Comment:
Related to the comment below and also matters for the other validations: try make them straight forward.